### PR TITLE
Implement Unix Domain Sockets

### PR DESCRIPTION
**Pull request by caleblloyd**
- added ./ci/server directory for files that need to be mounted into the server
- added ./ci/config for test configs
- added ignored file ./ci/mysqld to mount Unix Domain Socket into host
- implemented Unix Domain Socket from #118 
- added tests and docs

### CODE REVIEW

File: .ci/config/config.uds.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "Data": {
+      "ConnectionString": "server=./.ci/mysqld/mysqld.sock;user id=mysqltest;password='test;key=\"val';port=3306;database=mysqltest;UseCompression=true;Use Affected Rows=true",
```
Comment (bgrainger) [inline .ci/config/config.uds.json]: Did you want `UseCompression=true` here?

File: src/MySqlConnector/UnixEndPoint.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+
+namespace MySql.Data
+{
+	// copied from https://github.com/mono/mono/blob/master/mcs/class/Mono.Posix/Mono.Unix/UnixEndPoint.cs
```
Comment (bgrainger) [inline src/MySqlConnector/UnixEndPoint.cs]: It's probably appropriate to copy the full copyright and license statement from the original file, too.

File: src/MySqlConnector/UnixEndPoint.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
+﻿// copied from https://github.com/mono/mono/blob/master/mcs/class/Mono.Posix/Mono.Unix/UnixEndPoint.cs
+
+//
+// Mono.Unix.UnixEndPoint: EndPoint derived class for AF_UNIX family sockets.
```
Comment (caleblloyd) [inline src/MySqlConnector/UnixEndPoint.cs]: Added Mono's Copyright Notice

File: src/MySqlConnector/UnixEndPoint.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+
+namespace MySql.Data
+{
+
+	public class UnixEndPoint : EndPoint
```
Comment (caleblloyd) [inline src/MySqlConnector/UnixEndPoint.cs]: Removed Serializeable Attribute

File: .ci/config/config.uds.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "Data": {
+      "ConnectionString": "server=./.ci/mysqld/mysqld.sock;user id=mysqltest;password='test;key=\"val';database=mysqltest;Use Affected Rows=true",
```
Comment (caleblloyd) [inline .ci/config/config.uds.json]: Removed Compression (it was there by mistake)

**Pull request merged**